### PR TITLE
ADD tilt twist control (twist around z)

### PR DESCRIPTION
**Pull request by EwoudSmeur**
This commit implements tilt twist control for vehicles that rotate slow
in the z axis compared to the other axes (like tailsitters).

The attitude error is decomposed in a tilt and a twist part, and the
tilt error is compensated assuming that the rotation around z is zero
(very slow in comparison).

I can add the test to the other algebra tests if needed.

### CODE REVIEW

File: sw/airborne/math/pprz_algebra_float.c [hunk 1]
```diff
@@ -618,6 +618,39 @@ void float_quat_of_rmat(struct FloatQuat *q, struct FloatRMat *rm)
   }
 }
 
+#include <stdio.h>
```
Comment (gautierhattenberger) [inline sw/airborne/math/pprz_algebra_float.c:621]: to be removed

File: sw/airborne/math/pprz_algebra_float.c [hunk 2]
```diff
@@ -618,6 +618,39 @@ void float_quat_of_rmat(struct FloatQuat *q, struct FloatRMat *rm)
...
+/**
+ * @brief Tilt twist decomposition of a quaternion (z axis)
+ *
```
Comment (gautierhattenberger) [inline sw/airborne/math/pprz_algebra_float.c:624]: maybe provide a bit of mathematical details ? it is not so obvious what this function does

File: sw/airborne/math/pprz_algebra_float.h [hunk 3]
```diff
@@ -481,6 +481,9 @@ extern void float_quat_of_orientation_vect(struct FloatQuat *q, const struct Flo
 /// Quaternion from rotation matrix.
 extern void float_quat_of_rmat(struct FloatQuat *q, struct FloatRMat *rm);
 
+/// Tilt twist decomposition of quaternion
+void float_quat_tilt_twist(struct FloatQuat *tilt, struct FloatQuat *twist, struct FloatQuat *quat);
```
Comment (gautierhattenberger) [inline sw/airborne/math/pprz_algebra_float.h:485]: add `extern`, for consistency, it is technically not needed

**Review by gautierhattenberger — CHANGES_REQUESTED** [review]
the test binary file should be removed

**Comment by EwoudSmeur** [comment]
@gautierhattenberger thanks for your comments, sorry I guess I was a bit hasty pushing this! It is tested though 😇

**Review by gautierhattenberger — APPROVED** [review]
thanks

**Pull request merged**